File: server/s2game/src/main/java/com/spectopics/s2game/dto/serverPayloads/BattlesStartedPayload.java
package com.spectopics.s2game.dto.serverPayloads;

import com.spectopics.s2game.models.LobbyState;

public class BattlesStartedPayload {
    // Tells all the clients the battle stage has started.
    public LobbyState lobbyState;

    public BattlesStartedPayload(LobbyState lobbyState) {
        this.lobbyState = lobbyState;
    }
}

File: front-end/src/dto/serverPayloads/BattlesStartedPayload.js
import { LobbyState } from '../../models/LobbyState.js';

export class BattlesStartedPayload {
    constructor(lobbyState) {
        this.lobbyState = lobbyState;
    }

    static fromDict(data) {
        return new BattlesStartedPayload(LobbyState.fromDict(data.lobbyState));
    }
}
